client/rpc/src/lib.rs

```diff
@@ -28,6 +28,137 @@ use rustc_hex::ToHex;
 use pallet_evm::ExitReason;
 use sha3::{Digest, Keccak256};
 
+mod estimate_gas_binary {
+	use std::borrow::Cow;
+	use ethereum_types::{H160, H256, U256};
+	use pallet_evm::{
+		BackendT, EvmBasic, Vicinity, Config, StackExecutor,
+		Precompiles as PrecompilesT, ExitReason, ExitFatal,
+	};
+	use fc_rpc_core::types::{CallRequest, Bytes};
+	use jsonrpc_core::Error;
+	use crate::error_on_execution_failure;
+
+	pub struct Backend<'vicinity> {
+		_vicinity: &'vicinity Vicinity,
+	}
+
+	impl<'vicinity> Backend<'vicinity> {
+		pub fn new(_vicinity: &'vicinity Vicinity) -> Self {
+			Self { _vicinity }
+		}
+	}
+
+	impl<'vicinity> BackendT for Backend<'vicinity> {
+		fn gas_price(&self) -> U256 { U256::zero() }
+		fn origin(&self) -> H160 { H160::default() }
+		fn block_hash(&self, _number: U256) -> H256 { H256::default() }
+		fn block_number(&self) -> U256 { U256::zero() }
+		fn block_coinbase(&self) -> H160 { H160::default() }
+		fn block_timestamp(&self) -> U256 { U256::zero() }
+		fn block_difficulty(&self) -> U256 { U256::zero() }
+		fn block_gas_limit(&self) -> U256 { U256::zero() }
+		fn chain_id(&self) -> U256 { U256::zero() }
+		fn exists(&self, _address: H160) -> bool { true }
+		fn basic(&self, _address: H160) -> EvmBasic {
+			EvmBasic {
+				balance: U256::zero(),
+				nonce: U256::zero(),
+			}
+		}
+		fn code_size(&self, _address: H160) -> usize { 0 as usize }
+		fn code_hash(&self, _address: H160) -> H256 { H256::default() }
+		fn code(&self, _address: H160) -> Vec<u8> { Vec::new() }
+		fn storage(&self, _address: H160, _index: H256) -> H256 { H256::default() }
```

Comment:
This is not right..

If all those backend values are not properly fetched from the actual storage, the gas estimation will not be correct.

Reply:
Yeah, I was afraid of that. I have supercicial knowledge about the actual evm implementation: which of those defaulted values play a role in gas estimation?